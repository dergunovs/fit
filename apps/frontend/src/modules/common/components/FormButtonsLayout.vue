<template>
  <div>
    <div :class="$style.margin" v-if="props.isFixed" data-test="form-buttons-layout-margin"></div>

    <div :class="$style.buttons" :data-fixed="props.isFixed" data-test="form-buttons-layout-buttons">
      <slot></slot>
    </div>
  </div>
</template>

<script setup lang="ts">
interface IProps {
  isFixed?: boolean;
}

const props = defineProps<IProps>();
</script>

<style module lang="scss">
.margin {
  height: calc(var(--form-buttons-height) - 16px);
}

.buttons {
  display: flex;
  gap: 16px;
  justify-content: space-between;
  width: 100%;

  &[data-fixed='true'] {
    position: fixed;
    bottom: 54px;
    left: 0;
    z-index: 1;
    width: 100%;
    height: var(--form-buttons-height);
    padding: 12px;
    background-color: var(--color-white);
    border-top: 2px solid var(--color-gray);
  }
}
</style>
