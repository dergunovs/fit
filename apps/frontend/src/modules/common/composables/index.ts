import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
