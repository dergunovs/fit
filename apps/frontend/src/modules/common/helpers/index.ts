import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
