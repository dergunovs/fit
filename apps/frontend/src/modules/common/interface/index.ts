import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
