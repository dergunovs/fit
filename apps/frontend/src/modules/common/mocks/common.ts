import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
