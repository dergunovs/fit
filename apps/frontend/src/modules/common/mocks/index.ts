export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
