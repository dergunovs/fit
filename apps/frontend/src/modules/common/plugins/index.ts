import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
