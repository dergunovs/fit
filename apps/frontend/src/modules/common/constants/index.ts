import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
