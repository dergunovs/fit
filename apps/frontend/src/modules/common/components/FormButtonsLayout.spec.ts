import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import FormButtonsLayout from './FormButtonsLayout.vue';

import { wrapperFactory } from '@/common/test';

const margin = dataTest('form-buttons-layout-margin');
const buttons = dataTest('form-buttons-layout-buttons');

let wrapper: VueWrapper<InstanceType<typeof FormButtonsLayout>>;

beforeEach(() => {
  wrapper = wrapperFactory(FormButtonsLayout, { isFixed: true });
});

enableAutoUnmount(afterEach);

describe('FormButtonsLayout', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(FormButtonsLayout)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows margin if props isFixed', async () => {
    expect(wrapper.find(margin).exists()).toBe(true);

    await wrapper.setProps({ isFixed: false });

    expect(wrapper.find(margin).exists()).toBe(false);
  });

  it('sets fixed attribute to buttons styles', async () => {
    expect(wrapper.find(buttons).attributes('data-fixed')).toBe('true');

    await wrapper.setProps({ isFixed: false });

    expect(wrapper.find(buttons).attributes('data-fixed')).toBe('false');
  });
});
