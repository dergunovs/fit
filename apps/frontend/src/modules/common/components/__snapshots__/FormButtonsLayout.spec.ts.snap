// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormButtonsLayout > matches snapshot 1`] = `
"<div>
  <div class="_margin_871dab" data-test="form-buttons-layout-margin"></div>
  <div class="_buttons_871dab" data-fixed="true" data-test="form-buttons-layout-buttons"></div>
</div>"
`;
